gensim/models/translation_matrix.py

```diff
@@ -0,0 +1,329 @@
+            `lang_vec`: word2vec model that extract word vector for lexicon
+            `lexicon`: the default is None, if it is not provided, the lexicon
+                    is all the lang_vec's word, i.e. lang_vec.vocab.keys()
+        Returns:
+            `Space` object for the lexicon
+        """
+        # `words` to store all the word that
+        # `mat` to store all the word vector for the word in 'words' list
+        words = []
+        mat = []
+        if lexicon is not None:
+            # if the lexicon is not provided, using the all the Keyedvectors's words as default
+            for item in lexicon:
+                words.append(item)
+                mat.append(lang_vec.syn0[lang_vec.vocab[item].index])
+
+        else:
+            for item in lang_vec.vocab.keys():
+                words.append(item)
+                mat.append(lang_vec.syn0[lang_vec.vocab[item].index])
+
+        return Space(mat, words)
+
+    def normalize(self):
+        """ normalized the word vector's matrix """
+        self.mat = self.mat / np.sqrt(np.sum(np.multiply(self.mat, self.mat), axis=1, keepdims=True))
+
+
+class TranslationMatrix(utils.SaveLoad):
+    """
+    Objects of this class realize the translation matrix which map the source language
+    to the target language.
+     The main methods are:
+
+    1. constructor,
+    2. the `train` method, which initialize everything needed to build a translation matrix
+    3. the `translate` method, which given new word and its vector representation,
+    we map it to the other language space by computing z = Wx, then return the
+    word whose representation is close to z.
+
+    the details use seen the notebook (translation_matrix.ipynb)
+
+    >>> transmat = TranslationMatrix(word_pair, source_lang_vec, target_lang_vec)
+    >>> transmat.train(word_pair)
+    >>> translated_word = transmat.translate(words, topn=3)
+
+    """
+    def __init__(self, word_pair, source_lang_vec, target_lang_vec, random_state=None):
+        """
+        Initialize the model from a list pair of `word_pair`. Each word_pair is tupe
+         with source language word and target language word.
+
+        Examples: [("one", "uno"), ("two", "due")]
+
+        Args:
+            `word_pair` (list): a list pair of words
+            `source_lang_vec` (KeyedVectors): a set of word vector of source language
+            `target_lang_vec` (KeyedVectors): a set of word vector of target language
+        """
+
+        if len(word_pair[0]) != 2:
+            raise ValueError("Each training data item must contain two different language words.")
+
+        self.source_word, self.target_word = zip(*word_pair)
+        self.source_lang_vec = source_lang_vec
+        self.target_lang_vec = target_lang_vec
+
+        self.random_state = utils.get_random_state(random_state)
+        self.translation_matrix = None
+        self.source_space = None
+        self.target_space = None
+
+    def train(self, word_pair):
```

Comment:
This `word_pair` argument, which I noted seemed odd above, doesn't seem to be consulted at all in the following code. (Only the word-pairs supplied at initialization have any effect.)

Reply:
As you said in the second comment, if training was combined with initialization, then this parameter would be used in `train` method.

Reply:
Looking at the code body of the `train()` method, I see no references to this `word_pair` parameter. Hence, the parameter is unused and (unless the code changes) unnecessary here. If the intent was that this be an alternate way to supply the list-of-word-pairs, the code would have to change to reflect that intent.

Reply:
Okay, I change the code where the `word_pairs` is optional  parameter in the initialization, and if provided, then `train` method will run followed.